### PR TITLE
Fix markasread and breadCrumbcomp key prop error

### LINKED ISSUE #623: bug: Progress Not Saved
**Bug Description:**
Following the latest stream titled "Testing in Production," a bug has emerged wherein any new progress made in the cohort is not being saved. The "Mark as Complete" button also fails to function properly. Despite temporarily displaying as "Mark as Incomplete" upon completion of video content, it reverts to "Mark as Complete" and restarts from the old timestamp upon reloading.

**Steps to Reproduce:**
1. Navigate to any video within the cohort.
2. Attempt to complete the video.
3. Notice that completion is temporary and reverts upon reloading.

**Expected Behavior:**
Upon completion of the content, a green tick mark should appear next to the video. Similarly, clicking the "Mark as Complete" button should exhibit consistent behavior.

**Additional Information:**
- Browser: Brave

### LINKED ISSUE #624: Extra comma in package.json file
**Describe the bug**
There is an extra comma on line number 112 of package.json which is causing "npm install" to fail while using docker-compose up command.

**To Reproduce**
1. clone repo
2. run docker-compose up

**Expected behavior**
it should run without any issues to setup the project. It does when that comma is removed.

**Screenshots or GIFs**
image

### LINKED ISSUE #629: bug: Error in Brower console
**Describe the bug**

1-Extra attributes form the server: class,style;
2-Each child in a list should have a unique "key" prop breadCrumbComponent
3-Invaild Dom Property " stroke-whidh  "

**To Reproduce**
1. See in inspect mode in console

**Screenshots or GIFs**

1-Extra attributes form the server: class,style;
console log supress

2-Each child in a list should have a unique "key" prop breadCrumbComponent
brudcram

3-Invaild Dom Property " stroke-whidh  "
invaildom propertry

### LINKED ISSUE #630: bug: Husky Install comment is Deprecated
**Describe the bug**
Husky Install comment is Deprecated

**To Reproduce**
1. npm intall
2. Husky Install comment is Deprecated

**Screenshots or GIFs**

huskydepricated

### LINKED ISSUE #662: bug: Bookmarks seem broken
1. All of my bookmarks seemed as marked 
2. In actuality I have only marked a few
3. They are redirecting to the wrong place

Screenshot 2024-05-17 at 6 15 46 PM
Screenshot 2024-05-17 at 6 15 41 PM
Screenshot 2024-05-17 at 6 15 34 PM

### PR DESCRIPTION
**Pull request by manishpali17**
### PR Fixes:
- 1 Mark as complete feature fixed
- 2 Husky Install comment is Deprecated
- 3 Each child in a list should have a unique "key" prop breadCrumbComponent
- 4 Extra comma in package.json file

Resolves #630
Resolves #629 
Resolves #624
Resolves #623
Resolves #662 

### Checklist before requesting a review
- [x] I have performed a self-review of my code
- [x] I assure there is no similar/duplicate pull request regarding same issue

### CODE REVIEW

File: src/app/layout.tsx [hunk 1]
```diff
@@ -20,7 +20,7 @@ export const metadata: Metadata = siteConfig;
 
 export default function RootLayout({ children }: { children: ReactNode }) {
   return (
-    <html lang="en">
+    <html lang="en" suppressHydrationWarning>
```
Comment (manishpali17) [inline src/app/layout.tsx:23]: Used for browser console error Extra attributes form the server: class,style;

File: src/app/not-found.tsx [hunk 2]
```diff
@@ -18,9 +18,9 @@ export default function GlobalError() {
               stroke="currentColor"
             >
               <path
-                stroke-linecap="round"
-                stroke-linejoin="round"
-                stroke-width="2"
+                strokeLinecap="round"
+                strokeLinejoin="round"
+                strokeWidth="2"
```
Comment (manishpali17) [inline src/app/not-found.tsx:23]: Invaild Dom Property

File: src/components/BreadCrumbComponent.tsx [hunk 3]
```diff
@@ -88,30 +88,26 @@ export default function BreadCrumbComponent({
...
-                  </>
-                ) : (
-                  <BreadcrumbItem>
-                    <BreadcrumbPage className="font-semibold">
-                      {breadcrumb.title}
-                    </BreadcrumbPage>
-                  </BreadcrumbItem>
-                )}
-              </>
+            return index !== array.length - 1 ? (
```
Comment (manishpali17) [inline src/components/BreadCrumbComponent.tsx:91]: Each child in a list should have a unique "key" prop

File: src/components/CourseCard.tsx [hunk 4]
```diff
@@ -4,6 +4,8 @@ import PercentageComplete from './PercentageComplete';
 import { PrimaryButton } from './buttons/PrimaryButton';
 import { SecondaryButton } from './buttons/SecondaryButton';
 import { useRouter } from 'next/navigation';
+import { Button } from './ui/button';
+import { ChevronRight } from 'lucide-react';
```
Comment (manishpali17) [inline src/components/CourseCard.tsx:8]: missing imports

File: package.json [hunk 5]
```diff
@@ -19,7 +19,7 @@
     "prisma:migrate": "prisma migrate deploy",
     "db:seed": "prisma db seed",
     "db:reset": "prisma migrate reset",
-    "prepare": "husky install",
```
Comment (manishpali17) [inline package.json:22]: Husky Install comment is Deprecated It's "husky" now

File: src/db/course.ts [hunk 6]
```diff
@@ -173,11 +173,11 @@ export const getNextVideo = async (currentVideoId: number) => {
   return latestContent;
 };
 
-async function getAllContent() {
-  const value = Cache.getInstance().get('getAllContent', []);
-  if (value) {
-    return value;
-  }
+async function getAllContent(userId: string) {
+  //   const value = Cache.getInstance().get('getAllContent', []);
```
Comment (hkirat) [inline src/db/course.ts:177]: why this change?
